packages/compiler/lib/src/compilation/ast_hir_lowering/body.dart

```diff
@@ -740,6 +776,32 @@ class ExpressionContext extends InnerContext {
   }
 }
 
+class IfContext extends InnerContext {
+  IfContext(
+    Context parent,
+    this.id,
+    this.label,
+  )   : assert(id != null),
+        assert(label != null),
+        super(parent);
+
+  final DeclarationLocalId id;
+  final Option<String> label;
```

Comment:
```suggestion
  IfContext(
    Context parent,
    this.id,
  )   : assert(id != null),
        super(parent);

  final DeclarationLocalId id;
```

Reply:
We decided to leave the label and instead catch returns targeted at ifs.